azuread/table_azuread_application.go

```diff
@@ -35,145 +43,214 @@ func tableAzureAdApplication() *plugin.Table {
-	client.BaseClient.DisableRetries = true
-
-	application, _, err := client.Get(ctx, applicationId, odata.Query{})
+	application, err := client.ApplicationsById(applicationId).Get()
 	if err != nil {
-		if isNotFoundError(err){
-			return nil, nil
-		}
-		return nil, err
+		errObj := getErrorObject(err)
+		return nil, errObj
 	}
-	return *application, nil
+	isAuthorizationServiceEnabled := application.GetAdditionalData()["isAuthorizationServiceEnabled"]
+
+	return &ADApplicationInfo{application, isAuthorizationServiceEnabled}, nil
 }
 
-func getApplicationOwners(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
-	application := h.Item.(msgraph.Application)
-	session, err := GetNewSession(ctx, d)
+func getAdApplicationOwners(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	// Create client
+	client, adapter, err := GetGraphClient(ctx, d)
 	if err != nil {
-		if isNotFoundError(err) {
-			return nil, nil
-		}
-		return nil, err
+		return nil, fmt.Errorf("error creating client: %v", err)
+	}
+
+	application := h.Item.(*ADApplicationInfo)
+	applicationID := application.GetId()
+
+	headers := map[string]string{
+		"ConsistencyLevel": "eventual",
 	}
 
-	client := msgraph.NewApplicationsClient(session.TenantID)
-	client.BaseClient.Authorizer = session.Authorizer
+	includeCount := true
+	requestParameters := &owners.OwnersRequestBuilderGetQueryParameters{
+		Count: &includeCount,
+	}
 
-	owners, _, err := client.ListOwners(ctx, *application.ID)
+	config := &owners.OwnersRequestBuilderGetRequestConfiguration{
+		Headers:         headers,
+		QueryParameters: requestParameters,
+	}
+
+	ownerIds := []*string{}
+	owners, err := client.ApplicationsById(*applicationID).Owners().GetWithRequestConfigurationAndResponseHandler(config, nil)
+	if err != nil {
+		errObj := getErrorObject(err)
+		return nil, errObj
+	}
+
+	pageIterator, err := msgraphcore.NewPageIterator(owners, adapter, models.CreateDirectoryObjectCollectionResponseFromDiscriminatorValue)
+	err = pageIterator.Iterate(func(pageItem interface{}) bool {
+		owner := pageItem.(models.DirectoryObjectable)
+		ownerIds = append(ownerIds, owner.GetId())
+
+		return true
+	})
 	if err != nil {
 		return nil, err
 	}
-	return owners, nil
+
+	return ownerIds, nil
 }
 
-//// Transform Function
+//// TRANSFORM FUNCTIONS
+
+func adApplicationTags(ctx context.Context, d *transform.TransformData) (interface{}, error) {
+	application := d.HydrateItem.(*ADApplicationInfo)
+	tags := application.GetTags()
+	return TagsToMap(tags)
+}
+
+func adApplicationTitle(_ context.Context, d *transform.TransformData) (interface{}, error) {
+	data := d.HydrateItem.(*ADApplicationInfo)
+	if data == nil {
+		return nil, nil
+	}
+
+	title := data.GetDisplayName()
+	if title == nil {
+		title = data.GetId()
+	}
+
+	return title, nil
+}
+
+func buildApplicationQueryFilter(equalQuals plugin.KeyColumnEqualsQualMap) []string {
```

Comment:
I noticed some transform functions are in the table files, while others are in `helpers.go`. How did you decide where transform functions should live? Are some shared between different tables?

Reply:
Maybe rename to `transforms.go`?